### PR TITLE
v1.6.0 release notes and docs

### PR DESCRIPTION
**Pull request by ritch**
## Summary by CodeRabbit

- **New Features**
	- Added group management and Central Authentication Service for streamlined user authentication.
	- Introduced Pluggable Authentication system with legacy and internal modes.
	- Removed Auth0 dependency, enabling direct authentication with Identity Providers.
	- Launched a Super Admin UI and customizable login page.
	- Implemented support for user search functionality.
	- Enhanced roles, groups, and access permissions configuration.
	- Updated tutorials with Hugging Face Transformers Integration and Ultralytics Integration for YOLOv8.
	- Enhanced user groups for collective user management and dataset access control.

### CODE REVIEW

File: docs/source/release-notes.rst [hunk 1]
```diff
@@ -3,6 +3,22 @@ FiftyOne Release Notes
 
 .. default-role:: code
 
+FiftyOne Teams 1.6.0
+--------------------
+*Released March 29, 2024*
+
+What's New
```
Comment (ritch) [inline docs/source/release-notes.rst:10]: I'm going to add links throughout this section to the CAS doc. I will also need to add more content that matches to do so.

File: docs/source/teams/pluggable_auth.md [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+2. Assigning the user to the default organization through a membership entry.
+3. Linking the user's account to the new authentication provider by creating an
+   account reference via `POST /cas/api/accounts`.
+
+The final step in the migration involves changing the `FIFTYONE_AUTH_MODE`
+setting from legacy to internal. This change officially activates Internal
+Mode, completing the migration process.
+-->
+
+## Getting started with Internal Mode
```
Comment (manivoxel51) [inline docs/source/teams/pluggable_auth.md:129]: Maybe we add a note here about invitations not working yet. but saying you can still manage users through the super admin dashboard and rest api / js hooks.

**Review by manivoxel51 — DISMISSED** [review]
nice work! 🍨 

- one inline comment about invitations and internal mode
- In legacy mode, `FEATURE_FLAG_ENABLE_INVITATIONS` flag is live - not sure if we want to add it to docs.

File: docs/source/teams/pluggable_auth.md [hunk 3]
```diff
@@ -0,0 +1,668 @@
...
+   <td><strong>Description</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><code>services.util.http</code>
+   </td>
+   <td>Provides <code>get</code>, <code>post</code>, <code>put</code>, and <code>delete</code> functions for making <code>HTTP</code> requests from a JS Hook.
+   </td>
+  </tr>
+  <tr>
```
Comment (imanjra) [inline docs/source/teams/pluggable_auth.md:614]:
```suggestion
  <tr>
   <td><strong>Service</strong>
   </td>
   <td><strong>Description</strong>
   </td>
  </tr>
  <tr>
   <td><code>services.util.jose</code>
   </td>
   <td>Provides various JWT utilities such as <code>jwtVerify</code>, <code>decodeJwt</code>, <code>SignJWT</code>, and much more for working with JWT tokens.
   </td>
  </tr>
  <tr>
```

File: docs/source/teams/pluggable_auth.md [hunk 4]
```diff
@@ -0,0 +1,668 @@
+(pluggable-auth)=
```
Comment (imanjra) [inline docs/source/teams/pluggable_auth.md:1]: Minor suggestion (may ignore): should we rename this file to `.rst`

Comment (ritch) [reply docs/source/teams/pluggable_auth.md:1]: Hmm? Its markdown I don't think renaming it would be a good idea.

Comment (kevin-dimichel) [reply docs/source/teams/pluggable_auth.md:1]: It appears that all of the other documents are in Restructured Text (`.rst`) format, not Markdown (`.md`).

```
[fiftyone]$ find docs/source/teams -type f -name '*.rst'
docs/source/teams/index.rst
docs/source/teams/teams_app.rst
docs/source/teams/teams_plugins.rst
docs/source/teams/migrations.rst
docs/source/teams/api_connection.rst
docs/source/teams/overview.rst
docs/source/teams/dataset_versioning.rst
docs/source/teams/cloud_media.rst
docs/source/teams/installation.rst
docs/source/teams/management_sdk.rst
docs/source/teams/roles_and_permissions.rst
docs/source/teams/secrets.rst
[fiftyone]$ find docs/source/teams -type f -name '*.md'
docs/source/teams/pluggable_auth.md
```

Comment (brimoor) [reply docs/source/teams/pluggable_auth.md:1]: Also sets a precedent that more files can be contributed as `.md`, which would be sad to my OCD as they do render slightly differently.

It feels like we might be close to a major docs overhaul (moving to a new platform) so perhaps the inconsistency won't have much time to take hold.

File: docs/source/teams/pluggable_auth.md [hunk 5]
```diff
@@ -0,0 +1,683 @@
...
+
+> **NOTE**: The value of `FIFTYONE_AUTH_SECRET` should be set prior to
+> installation or upgrade.
+
+(identity-providers)=
+
+## Identity Providers (IdP)
+
+In [internal mode](#internal-mode), use the CAS REST API or Super Admin UI to
+configure FiftyOne teams to authenticate users via OIDC or OAuth2 compatible
```
Comment (kevin-dimichel) [inline docs/source/teams/pluggable_auth.md:78]: I am under the impression that OAuth2 is a protocol for authorization, not authentication.  Should we update this to be 

```suggestion
configure FiftyOne teams to authenticate users via OIDC compatible
```

?

**Comment by findtopher** [comment]
I'm linking the install instructions to the [Migrating from Legacy to internal mode](https://github.com/voxel51/fiftyone/blob/release-notes%2Fv1.6.0/docs/source/teams/pluggable_auth.md?plain=1#L114) header for migrating from legacy-to-internal mode (assumption is `https://docs.voxel51.com/teams/pluggable_auth.html#migrating-from-legacy-to-internal-mode`)

Please let me know when that section is complete and if the title changes!

File: docs/source/teams/pluggable_auth.md [hunk 6]
```diff
@@ -0,0 +1,683 @@
...
+FiftyOneTeams eventually). Auth0 contains the configuration for identity
+providers and the persistence of user data. Auth0 supports multiple providers
+(including SAML). For the supported IdPs, see
+[Auth0 Enterprise Identity Providers](https://auth0.com/docs/authenticate/identity-providers/enterprise-identity-providers).
+
+(internal-mode)=
+
+### Internal Mode
+
+In internal mode, FiftyOne Teams the CAS replaces Auth0. FiftyOne Teams will
```
Comment (allenleetc) [inline docs/source/teams/pluggable_auth.md:34]: Should this be "... the FiftyOne Teams CAS ..."

File: docs/source/teams/pluggable_auth.md [hunk 7]
```diff
@@ -0,0 +1,683 @@
...
+    1. In the “Profile callback” field, set the mapping that your Identity
+       Provider expects
+1. Login with the admin user
+    1. Navigate to `https://≤YOUR_FIFTYONE_TEAMS_URL>/datasets`
+    1. You should see the login screen for your newly configured authentication
+       provider
+    1. Before logging in, set the admin user (in step 5). Otherwise, you will
+       need to remove this user from the database and try again.
+    1. Click the login button and provide the credentials of the Admin user
+       (set in step 5)
```
Comment (allenleetc) [inline docs/source/teams/pluggable_auth.md:164]: Here should the "in step 5" be "in step 3"

Comment (kevin-dimichel) [reply docs/source/teams/pluggable_auth.md:164]:
```suggestion
       (set in step 3)
```

File: docs/source/teams/pluggable_auth.md (hunk 5 above)
Comment (ritch) [reply docs/source/teams/pluggable_auth.md:78]: No this is true, the settings we expose are a combination of both protocols, regardless of the semantics there are literally OAuth and OIDC options. This is also based on what we use: next-auth which is self-described as such:

> NextAuth.js is designed to work with any OAuth service, it supports OAuth 1.0, 1.0A, 2.0 and OpenID Connect (OIDC) and has built-in support for most popular sign-in services.

**Review by manivoxel51 — APPROVED** [review]
thank you! 🍨

**Review by brimoor — APPROVED** [review]

**Pull request merged**